demonstrations/tutorial_general_parshift.py

```diff
@@ -0,0 +1,880 @@
+depend on a single parameter. We will investigate the form such functions usually take
+and demonstrate how we can *reconstruct* them as classical functions, capturing the full
+dependence on the input parameter.
+Once we have this reconstruction, we use it to compute analytically exact derivatives
+of the quantum function. We implement this in two ways:
+first, by using autodifferentiation on the classical function that is produced by the 
+reconstruction, which is flexible with respect to the degree of the derivative.
+Second, by computing the derivative manually, resulting in generalized parameter-shift
+rules for quantum functions that is more efficient (regarding classical cost) than the 
+autodifferentiation approach, but requires manual computations if we want to access
+higher-order derivatives.
+All we will need for the demo is the insight that these functions are Fourier series in their
+variable, and the reconstruction itself is a
+`trigonometric interpolation <https://en.wikipedia.org/wiki/Trigonometric_interpolation>`_.
+
+A full description of the reconstruction, the technical derivation of the parameter-shift
+rules, and considerations for multivariate functions can be found in the paper
+`General parameter-shift rules for quantum gradients <https://arxiv.org/abs/2107.12390>`_
+[#GenPar]_.
+The core idea to consider these quantum functions as Fourier series was first presented in
+the preprint
+`Calculus on parameterized quantum circuits <https://arxiv.org/abs/1812.06323>`_ [#CalcPQC]_.
+We will follow [#GenPar]_, but there also are two preprints discussing general parameter-shift
+rules: an algebraic approach in
+`Analytic gradients in variational quantum algorithms: Algebraic extensions of the parameter-shift rule to general unitary transformations <https://arxiv.org/abs/2107.08131>`_ [#AlgeShift]_
+and one focusing on special gates and spectral decompositions, namely
+`Generalized quantum circuit differentiation rules <https://arxiv.org/abs/2108.01218>`_
+[#GenDiffRules]_.
+
+|
+
+.. figure:: ../demonstrations/general_parshift/thumbnail_genpar.png
+    :align: center
+    :width: 50%
+    :target: javascript:void(0)
+
+    Function reconstruction and differentiation via parameter shifts.
+
+.. note ::
+
+    Before going through this tutorial, we recommend that readers refer to the
+    :doc:`Fourier series expressiveness tutorial </demos/tutorial_expressivity_fourier_series>`.
+    Additionally, having a basic understanding of the
+    :doc:`parameter-shift rule </glossary/parameter_shift>` might make this tutorial easier
+    to dive into.
+
+Cost functions arising from quantum gates
+-----------------------------------------
+We start our investigations with a cost function that arises from measuring the expectation
+value of an observable in a quantum state created with a parametrized quantum operation
+that depends on a single variational parameter :math:`x`.
```

Comment:
```suggestion
value of an observable on a quantum state, created with a parametrized quantum operation
that depends on a single variational parameter :math:`x`.
```
I'm not 100% sure here, but is it on or in?

Reply:
I'd favor "in".